File: leetcode/C++/valid-mountain-array.cpp
class Solution {
public:
    bool validMountainArray(vector<int>& arr) {
        int inIdx=0, deIdx=(int)arr.size()-1;
        for(; inIdx+1 < arr.size(); ++inIdx) if(arr[inIdx]>=arr[inIdx+1]) break;
        for(; deIdx >= 1; --deIdx) if(arr[deIdx]>=arr[deIdx-1]) break;
        return (inIdx==deIdx) && (inIdx>0) && (inIdx+1<arr.size());
    }
};

File: leetcode/Python/valid-mountain-array.py
class Solution:
    def validMountainArray(self, arr: List[int]) -> bool:
        inIdx, deIdx = 0, len(arr)-1
        while inIdx + 1 < len(arr):
            if arr[inIdx] >= arr[inIdx+1]:
                break
            inIdx += 1
        while deIdx >= 1:
            if arr[deIdx] >= arr[deIdx-1]:
                break
            deIdx -= 1
        return inIdx==deIdx and inIdx > 0 and inIdx+1 < len(arr)